### DISCUSSION TITLE
lte V2X communication with custom ns3 library

### DISCUSSION THREAD

**Discussion by eqwhip** [eclipse/mosaic#226] [Q&A]
Hello,

I am trying to implement lte-v2x sidelink communication using a custom ns3 library.

In order to do that I would need to import that custom library to mosaic and set the parameters. I tried tweaking the ns3_federate_config.xml file, but it resulted in errors, as I assume it is only for 802.11(p) settings.

![image](https://user-images.githubusercontent.com/15569305/172873524-3924f9c2-287e-466e-8a11-54afa65feec3.png)

![image](https://user-images.githubusercontent.com/15569305/172873784-cc06586d-32dd-48bf-b609-5b43fbb0f1ab.png)

![image](https://user-images.githubusercontent.com/15569305/172874033-891d53ec-2d93-4cf4-a7dc-2fc53c8fb8ce.png)

_note: I have used tiergarten scenario, and from default scenario_config.json changed the ns3 federator option true, and sns as false._

So, is there any solution or method (Even a lengthy one) to integrate custom NS3 lte-v2x model based messages inside the Eclipse Mosaic simulation.

Any inputs/suggestions will be of great help, thank you.

**Comment by Cheng-Yuan-0124** [comment]
@eqwhip Hello, I currently have a similar project, would you mind if you can share some information with me
